go/vt/discovery/tablet_picker.go

```diff
@@ -146,6 +146,9 @@ func NewTabletPicker(
 	options TabletPickerOptions,
 ) (*TabletPicker, error) {
 	// Keep inOrder parsing here for backward compatability until TabletPickerTabletOrder is fully adopted.
+	if tabletTypesStr == "" {
```

Comment:
This looks like a genuine bug, or is this a side-effect of changes in this PR? If no tablet types were specified in SwitchTraffic, we would fail because the picker didn't find any tablet types to work on, rather than default to all.

Reply:
We should always be specifying tablet types -- the default should be applied externally. What did you see fail?

Reply:
Forget now, but one of the tests had failed and digging into this I had found a code path that ended up with no string being defined here. Maybe that was due to some other issue at that time, we can remove that and check the tests. In any case, I guess this is defensive code that should not harm.